src/base_alloc/base_alloc.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <assert.h>
+
+#include <base_alloc.h>
+#include <base_alloc_linear.h>
+#include <base_alloc_os_mem.h>
+
+// pool size of the base allocator,
+// it will be reallocated if we run out of memory.
+#define BA_POOL_SIZE (ba_os_get_page_size())
+
+// alignment of the base allocator
+#define MEMORY_ALIGNMENT (8)
+
+typedef struct ba_list_item_t ba_list_item_t;
+
+struct ba_list_item_t {
+    ba_list_item_t *next;
+    void *addr;
+};
+
+struct umf_ba_pool_t {
+    size_t aligned_size;
+    ba_list_item_t *free_list;
+    ba_list_item_t *alloc_list;
+#ifdef DEBUG
+    long tid;
+#endif /* DEBUG */
+};
+
+static int ba_fill_pool(umf_ba_pool_t *pool) {
```

Comment:
add a one-sentence description what this function does

Reply:
Done